### PR TITLE
Feature/change published proposal to enrollment closed

### PR DESCRIPTION
**Pull request by ManuPerezDev**
This is a feature so that the contact person can update the proposal status from PUBLISHED to ENROLLMENT_CLOSED.

### CODE REVIEW

File: backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java [hunk 1]
```diff
@@ -107,4 +110,23 @@ public ProposalRevisionEmail requestChanges(
     private boolean hasFeedback(ProposalRevisionDto proposalRevisionDto) {
         return proposalRevisionDto.getFeedback() != null;
     }
+
+    /**
+     * This method find the proposal in the database and checks if the status is PUBLISHED for
+     * close the enrollment. Otherwise, a ProposalNotClosableException with response status 409 will be throw.
+     * @param idProposal : The id of the proposal to be checked and updated.
+     */
+    public void changeStatusToEnrollmentClosed(String idProposal) {
```
Comment (ibaiul) [inline backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java]: What about `closeEnrollment(String proposalId)`?

File: backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java [hunk 2]
```diff
@@ -107,4 +110,23 @@ public ProposalRevisionEmail requestChanges(
...
+        JpaProposal proposal = jpaProposalRepository.findByNaturalId(idProposal).orElseThrow(EntityNotFoundException::new);
+        String status = proposal.getStatus().getName().toUpperCase();
+
+        if (!PUBLISHED.toString().equals(status)) {
+            throw new ProposalNotClosableException();
```
Comment (ibaiul) [inline backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java]: The proposal does not get closed. What it does get closed is the enrollment of the proposal.

ProposalEnrollmentNotCloseableException/EnrollmentNotCloseableException

File: backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java [hunk 3]
```diff
@@ -107,4 +110,23 @@ public ProposalRevisionEmail requestChanges(
     private boolean hasFeedback(ProposalRevisionDto proposalRevisionDto) {
         return proposalRevisionDto.getFeedback() != null;
     }
+
+    /**
+     * This method find the proposal in the database and checks if the status is PUBLISHED for
+     * close the enrollment. Otherwise, a ProposalNotClosableException with response status 409 will be throw.
```
Comment (ibaiul) [inline backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java]: Too many technical details. You don't need to explain the code below in the Javadoc.

```
Close the enrollment of the provided proposal.
@param ...
@throws XXXException when the proposal is not published.
```

File: backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java (hunk 1 above)
Comment (ManuPerezDev) [reply backend/src/main/java/com/huellapositiva/domain/service/ProposalService.java]: Got it! I think it fits better. All help with naming is welcome. I'm a mess with naming!

**Review by ibaiul — DISMISSED** [review]

**Review by ibaiul — APPROVED** [review]

**Pull request merged**